iree/vm/bytecode_module.c

```diff
@@ -649,19 +665,34 @@ static iree_status_t iree_vm_bytecode_module_begin_call(
                             module->function_descriptor_count);
   }
 
-  // Enter function (as this is the initial call).
-  // The callee's return will take care of storing the output registers when it
-  // actually does return, either immediately or in the future via a resume.
-  iree_vm_stack_frame_t* callee_frame = NULL;
-  IREE_RETURN_IF_ERROR(
-      iree_vm_stack_function_enter(stack, function, call->argument_registers,
-                                   call->result_registers, &callee_frame));
+  // Grab calling convention string. This is not great as we are guaranteed to
+  // have a bunch of cache misses, but without putting it on the descriptor
+  // (which would duplicate data and slow down normal intra-module calls)
+  // there's not a good way around it. In the grand scheme of things users
+  // should be keeping their calls across this boundary relatively fat (compared
+  // to the real work they do), so this only needs to be fast enough to blend
+  // into the noise.
```

Comment:
Good points here, thanks for the comment. I'm in agreement that calls across the boundary should be handled carefully. Is there anything more we can do here to help these costs appear clearly under profiling? We can also make sure any of our samples demonstrate best practices well. Maybe we could even put together some counter-examples or optimization walkthroughs ("here's poor API usage and a trace file", "if you make these changes, the trace and benchmarks now look like this instead").